File: gaia-sdk-javascript/src/graphql/request/type/DeletedApiKeyImpulse.ts

import {ApiKeyKeyOne} from "./ApiKeyKeyOne";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Impulse which indicates the result of a delete api key impulse
 */
export class DeletedApiKeyImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "DeletedApiKeyImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    public data = (config: (_:ApiKeyKeyOne) => void) => this.push((registry) => {
        const entity = new ApiKeyKeyOne();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/response/type/DeletedApiKeyImpulse.py

from gaia_sdk.graphql.response.type.ApiKeyKeyOne import ApiKeyKeyOne

Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool


class DeletedApiKeyImpulse:
    """
    Impulse which indicates the result of a delete api key impulse
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    @property
    def id(self) -> Uuid:
        return Uuid(self.dictionary.get("id"))
    @property
    def data(self) -> ApiKeyKeyOne:
        return ApiKeyKeyOne(self.dictionary.get("data"))
